### PR TITLE
Improve RGI database handling

### PR DESCRIPTION
**Pull request by jasmezz**
RGI database follows `--save-databases` logic and can also be supplied by user now (see changelog and usage.md).

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/funcscan/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/funcscan _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,singularity --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by jfy133 — COMMENTED** [review]
I don't like this copy to function... What if a user has set publishing to outdir via symlink. Why do you prefer supplying the link directly versus the Card annotation module that you deleted?

**Comment by jasmezz** [comment]
The `cardannotation` module is still executed, I just deleted its database output because it outputs a modified version, not the original one. But maybe it doesn't matter and the user can just re-input the modified one. Let me test that, then we could go back and get rid of my current copyTo solution.

File: nextflow_schema.json [hunk 1]
```diff
@@ -847,6 +847,12 @@
                     "description": "Skip RGI during the ARG-screening.",
                     "fa_icon": "fas fa-ban"
                 },
+                "arg_rgi_database": {
+                    "type": "string",
+                    "description": "Path to user-defined local CARD database.",
+                    "fa_icon": "fas fa-layer-group",
+                    "help_text": "You can pre-download the CARD database to your machine and pass the path of it to this parameter.\n\nSee the pipeline [documentation](https://nf-co.re/funcscan/usage#rgi) for details on how to download this."
```
Comment (jfy133) [inline nextflow_schema.json:854]: What parameter does RGI use to accept the user-supplied database (if any)

File: subworkflows/local/arg.nf [hunk 2]
```diff
@@ -85,10 +85,21 @@ workflow ARG {
...
     if ( !params.arg_skip_rgi ) {
 
-        // Download and prepare CARD
-        UNTAR ( [ [], file('https://card.mcmaster.ca/latest/data', checkIfExists: true).copyTo(params.outdir + '/databases/card/data.tar.gz') ] )
-        ch_versions = ch_versions.mix( UNTAR.out.versions )
-        RGI_CARDANNOTATION ( UNTAR.out.untar.map{ it[1] } )
+        if ( !params.arg_rgi_database ) {
+
+            // Download und untar CARD
+            UNTAR ( [ [], file('https://card.mcmaster.ca/latest/data', checkIfExists: true).copyTo("${params.outdir}/databases/rgi/data.tar.gz") ] )
```
Comment (jfy133) [inline subworkflows/local/arg.nf:91]: As the with the other I would just publish the UNTAR results rather than `copyTo`

**Review by jfy133 — COMMENTED** [review]
Almost there @jasmezz

**Review by jfy133 — APPROVED** [review]

**Pull request merged**